src/Hooks/useMutation.ts

```diff
@@ -1,52 +1,38 @@
-import { useState } from "react";
-const BASE_URL = process.env.REACT_APP_SERVER_URL;
+import { useState, useCallback } from "react";
 
-type useMutationParams = {
-  url: string;
-  method: string;
-};
+type mutationFuncType = (...args: any[]) => Promise<Response>;
 
-export function useMutation<T>(params: useMutationParams) {
-  const { url, method } = params;
+export function useMutation<T>(mutationFunc: mutationFuncType) {
+  const [status, setStatus] = useState<
+    "idle" | "loading" | "error" | "success"
```

Comment:
define a map of constants and expose these statuses. So, that user using these statuses for some condition can apply checks on the statuses without any typo